crates/cargo-codspeed/src/build.rs

```diff
@@ -122,13 +129,70 @@ impl BuildOptions<'_> {
         // In practice, if we set debug info through RUSTFLAGS, cargo still strips them, most
         // likely because debug = false in the release profile.
         // We also need to disable stripping through rust flags.
-        rust_flags.push_str(" -C strip=none");
+        flags.push("-C".to_string());
+        flags.push("strip=none".to_string());
 
         // Add the codspeed cfg flag if instrumentation mode is enabled
         if measurement_mode == MeasurementMode::Instrumentation {
-            rust_flags.push_str(" --cfg codspeed");
+            flags.push("--cfg".to_string());
+            flags.push("codspeed".to_string());
+        }
+
+        match std::env::var("RUSTFLAGS") {
+            std::result::Result::Ok(existing_rustflags) => {
+                // If RUSTFLAGS is set, append our flags to it
+                let mut combined_flags = existing_rustflags;
+                for flag in flags {
+                    combined_flags.push(' ');
+                    combined_flags.push_str(&flag);
+                }
+                cargo.env("RUSTFLAGS", combined_flags);
+            }
+            std::result::Result::Err(_) => {
+                // Check if cargo version supports --config (Cargo 1.63+, 2022-08-11)
+                if Self::cargo_supports_config_from_cli() {
+                    // Use --config to set rustflags
+                    let config_value = format!(
+                        "build.rustflags=[{}]",
+                        flags.into_iter().map(|f| format!("\"{f}\"")).join(",")
+                    );
+                    cargo.arg("--config").arg(config_value);
+                } else {
+                    // Fallback to RUSTFLAGS for older cargo versions
+                    let rustflags = flags.join(" ");
+                    cargo.env("RUSTFLAGS", rustflags);
+                }
+            }
         }
-        cargo.env("RUSTFLAGS", rust_flags);
+    }
+
+    /// Check if cargo version supports --config flag (Cargo 1.63+)
+    /// [https://doc.rust-lang.org/nightly/cargo/CHANGELOG.html#cargo-163-2022-08-11]
+    fn cargo_supports_config_from_cli() -> bool {
+        let output = Command::new("cargo").arg("--version").output();
+
+        if let std::result::Result::Ok(output) = output {
+            if let std::result::Result::Ok(version_str) = String::from_utf8(output.stdout) {
+                // Parse version string like "cargo 1.70.0 (7fe40dc9c 2023-04-27)"
+                if let Some(version_part) = version_str.split_whitespace().nth(1) {
+                    if let std::result::Result::Ok(version) = Version::parse(version_part) {
+                        // Cargo 1.63.0 introduced --config support
+                        return version >= Version::new(1, 63, 0);
+                    }
+                }
+            }
+        }
+
+        false
+    }
+
```

Comment:
Our MSRV is 1.74 so I don't think we need this check, maybe a comment mentioning this was introduced in 1.63 and our MSRV is greater than this will be enough